compiler/sem/sigmatch.nim

```diff
@@ -2608,7 +2612,7 @@ proc prepareOperand(c: PContext; formal: PType; a, aOrig: PNode): PNode =
   if formal.kind == tyUntyped:
     assert formal.len != 1
     result = aOrig
-  elif a.typ.isNil:
+  elif nfSem notin a.flags:
     # XXX This is unsound! 'formal' can differ from overloaded routine to
     # overloaded routine!
```

Comment:
Soft suggestion, but you can remove this if you'd like.

I don't think this is valid anymore, because of `untyped` vs non-`untyped` argument isolation, the soon to be included `nfSem` flag guards, and the removal of the `tyIterable` hack a while back.